linux/src/main/java/com/github/weisj/darkmode/platform/linux/gnome/GnomeThemeMonitorService.java

```diff
@@ -0,0 +1,82 @@
+package com.github.weisj.darkmode.platform.linux.gnome;
+
+import com.github.weisj.darkmode.platform.ThemeMonitorService;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.github.weisj.darkmode.platform.linux.gnome.GtkVariants.guessFrom;
+import static java.util.Objects.isNull;
+
+public class GnomeThemeMonitorService implements ThemeMonitorService {
+    private final Logger logger = Logger.getLogger(GnomeThemeMonitorService.class.getName());
+    private boolean initializedGnomeGtkThemeMonitorSuccessfully;
+    private GnomeGtkThemeMonitor gnomeGtkThemeMonitor;
+    private Thread gnomeGtkThemeMonitorThread;
+
+    public GnomeThemeMonitorService() {
+        try {
+            gnomeGtkThemeMonitor = new GnomeGtkThemeMonitor();
+            initializedGnomeGtkThemeMonitorSuccessfully = true;
+        } catch (IOException | InterruptedException e) {
+            initializedGnomeGtkThemeMonitorSuccessfully = false;
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public boolean isDarkThemeEnabled() {
+        //TODO: Stop guessing and check the settings when available (like what is mentioned in the GtkVariants class)
+        return gnomeGtkThemeMonitor.readGtkTheme().equals(guessFrom(gnomeGtkThemeMonitor.readGtkTheme()).get("night"));
+    }
+
+    @Override
+    public boolean isHighContrastEnabled() {
+        //TODO: This right now isn't exactly doable with the guessing implementation. It requires a user-accessible place to
+        // set which theme is their "high contrast theme"
+        return false;
+    }
+
+    @Override
+    public long createEventHandler(Runnable callback) {
+        try {
+            gnomeGtkThemeMonitor.setCallback(callback);
+        } catch (Exception e) {
+            logger.log(Level.SEVERE, e.getMessage(), e);
+            return 0;
+        }
+        gnomeGtkThemeMonitorThread = new Thread(gnomeGtkThemeMonitor);
+        gnomeGtkThemeMonitorThread.start();
+        return 1;
+    }
+
+    @Override
+    public void deleteEventHandler(long eventHandle) {
+        gnomeGtkThemeMonitor.deleteEventHandler(eventHandle);
+    }
+
+    @Override
+    public boolean isActive() {
```

Comment:
I have to admit that the name `isActive()` is unfortunately chosen. It's actually only called once to check whether the service can function appropriately. Maybe `isSupported()` would be a better name. Then the only value that is relevant is `initializedGnomeGtkThemeMonitorSuccessfully`.

Reply:
You are free to rename the method to a more suitable name.

Reply:
Fixed in e5934c2826d5491cd7839b0edd693fabaeaafa1d.